tests/test_parse_file.cpp

```diff
@@ -924,38 +920,36 @@ BOOST_AUTO_TEST_CASE(test_file_ends_without_lf)
 
 BOOST_AUTO_TEST_CASE(test_parse_function_compiles)
 {
+    const auto c = [](std::string& s) -> const std::string& { return s; };
+    /*mutable*/ std::string example = testinput("example.toml");
+
     // toml::parse("");
-    const auto string_literal = toml::parse("toml/tests/example.toml");
+    using result_type = decltype(toml::parse("string literal"));
 
     BOOST_TEST_MESSAGE("string_literal");
 
-    const char* fname_cstring = "toml/tests/example.toml";
     // toml::parse(const char*);
-    const auto cstring = toml::parse(fname_cstring);
+    const result_type cstring = toml::parse(example.c_str());
 
     BOOST_TEST_MESSAGE("const char*");
 
     // toml::parse(char*);
-    std::array<char, 24> fname_char_ptr;
-    std::strncpy(fname_char_ptr.data(), fname_cstring, 24);
-    const auto char_ptr = toml::parse(fname_char_ptr.data());
+    const result_type char_ptr = toml::parse(&example.front());
 
     BOOST_TEST_MESSAGE("char*");
 
     // toml::parse(const std::string&);
-    const std::string fname_string("toml/tests/example.toml");
-    const auto string = toml::parse(fname_string);
-    std::string fname_string_mut("toml/tests/example.toml");
+    const result_type string = toml::parse(c(example));
     // toml::parse(std::string&);
-    const auto string_mutref = toml::parse(fname_string_mut);
+    const result_type string_mutref = toml::parse(example);
     // toml::parse(std::string&&);
-    const auto string_rref = toml::parse(std::move(fname_string_mut));
+    const result_type string_rref = toml::parse(std::move(example));
 
     BOOST_TEST_MESSAGE("strings");
 
 #ifdef TOML11_HAS_STD_FILESYSTEM
-    const std::filesystem::path fname_path(fname_string.begin(), fname_string.end());
-    const auto filesystem_path = toml::parse(fname_path);
+    const std::filesystem::path fname_path(example.begin(), example.end());
```

Comment:
Currently some of the CI tasks are failing in this test case. I didn't look into it deeper, but at a glance, here you use a value example that is already moved at L946 (actually fname_string_mut is introduced to avoid this). I'm not sure whether this causes the error, but at least we need to fix this.

Reply:
Sorry for the somewhat late reply, I've been on a short vacation. I'll look into this problem as soon as the most urgent problems at work are resolved.

Reply:
I have to admit that I am very confused about the new build failure.

```
[ 53%] Built target test_parse_array
[ 54%] Building CXX object tests/CMakeFiles/test_parse_table.dir/test_parse_table.cpp.o
In file included from /home/runner/work/toml11/toml11/toml/value.hpp:8,
                 from /home/runner/work/toml11/toml11/toml/get.hpp:9,
                 from /home/runner/work/toml11/toml11/tests/test_parse_table.cpp:1:
/home/runner/work/toml11/toml11/toml/exception.hpp: In constructor ‘toml::file_io_error::file_io_error(int, const string&, const string&)’:
/home/runner/work/toml11/toml11/toml/exception.hpp:17:66: error: ‘strerror’ is not a member of ‘std’
   17 |         : std::runtime_error(msg + " \"" + fname + "\": " + std::strerror(errnum)),
      |                                                                  ^~~~~~~~
make[2]: *** [tests/CMakeFiles/test_parse_table.dir/build.make:76: tests/CMakeFiles/test_parse_table.dir/test_parse_table.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:666: tests/CMakeFiles/test_parse_table.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

Not only does it seem unrelated to the recent change, I also cannot find the tokens `strerror` or `file_io_error` anywhere in the code base.

Do you have any ideas what could have gone wrong here? I couldn't reproduce the failure locally either.

Reply:
That error is not your fault. Recently I merged another PR, so the updated patch is applied to the new code and caused the error.
I have just fixed it ~3 hours ago. Could you merge the current `master` branch to your `several-simple-patches` branch? Hopefully, it will fix the error.

Reply:
Thanks for the explanation, that really got me confused. I have rebased my branch onto the current `master` branch now.